packages/loader/container-loader/src/container.ts

```diff
@@ -1212,12 +1212,10 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             this._existing = details.existing;
         }
 
-        // LoadContext directly requires protocolHandler to be ready, and eventually calls
-        // instantiateRuntime which will want to know existing state.
         this._protocolHandler = await protocolHandlerP;
 
         const codeDetails = this.getCodeDetailsFromQuorum();
-        await this.loadContext(codeDetails, attributes, snapshot, pendingLocalState);
+        await this.createExistingContext(codeDetails, attributes, snapshot, pendingLocalState);
```

Comment:
This assumes that legacy path always gives us this._existing = true. If it's not the case, then it's not safe to go that route, right? If it's limited to some tool only (let's say snapshot tests), then I'd figure out some way to assert that invariant (maybe by passing a loader flag saying - disable asserting that we always loading from existing)